service/history/chasm/components/activity/fx.go

```diff
@@ -0,0 +1,51 @@
+package activity
+
+import (
+	"go.temporal.io/server/api/matchingservice/v1"
+	"go.temporal.io/server/service/history/chasm"
+	"go.uber.org/fx"
+)
+
+type Library struct {
+	matchingClient matchingservice.MatchingServiceClient
+}
+
+func (l Library) Name() string {
+	return "activity"
+}
+
+func (l Library) Components() []chasm.RegistrableComponent {
+	return []chasm.RegistrableComponent{
+		chasm.NewRegistrableComponent[*Activity](
```

Comment:
I'm kinda going back and forth between a `RegisterComponent` function and this approach where the library returns the list of components.

The main downside to this approach is that you have to wrap with `NewRegisterableComponent`.

If we do want this approach, we may want a `LibraryBase` struct that has empty implementations for all of the methods so library implementors only provide the items they need registered.

I like that with the concept of a library, you essentially get a namespace for all components and tasks.